megamek/src/megamek/server/Server.java

```diff
@@ -887,6 +887,16 @@ private void receivePlayerVersion(Packet packet, int connId) {
             System.out.println("ERROR: Client/Server Version Mismatch -- Client: "+version+" Server: "+MegaMek.VERSION);
             needs = true;
         }
+        else if (clientChecksum == null) {    // print a message indicating client doesn't have jar file
+            buf.append("Client Checksum is null. Client may not have a jar file");
+            System.out.println("ERROR: Client does not have a jar file");
+            needs = true;      	
+        }
+        else if (serverChecksum == null) {    // print message indicating server doesn't have jar file
+            buf.append("Server Checksum is null. Server may not have a jar file");
+            System.out.println("ERROR: Server does not have a jar file");
+            needs = true; 
+        }
         if (!clientChecksum.equals(serverChecksum)) {
```

Comment:
This needs to be an else-if with the previous conditional block, otherwise you won't actually prevent the null pointer exception.